test/DGmethods/compressible_Navier_Stokes/rising_bubble-model-imex.jl

```diff
@@ -0,0 +1,220 @@
+journal = {Journal of the Atmospheric Sciences},
+volume = {50},
+number = {13},
+pages = {1865-1873},
+year = {1993},
+doi = {10.1175/1520-0469(1993)050<1865:BCEWAS>2.0.CO;2},
+URL = {https://doi.org/10.1175/1520-0469(1993)050<1865:BCEWAS>2.0.CO;2},
+eprint = {https://doi.org/10.1175/1520-0469(1993)050<1865:BCEWAS>2.0.CO;2},
+}
+"""
+function Initialise_Rising_Bubble!(state::Vars, aux::Vars, (x1,x2,x3), t)
+  DF            = eltype(state)
+  R_gas::DF     = R_d
+  c_p::DF       = cp_d
+  c_v::DF       = cv_d
+  γ::DF         = c_p / c_v
+  p0::DF        = MSLP
+  
+  xc::DF        = 500
+  zc::DF        = 260
+  r             = sqrt((x1 - xc)^2 + (x3 - zc)^2)
+  rc::DF        = 250
+  θ_ref::DF     = 303
+  Δθ::DF        = 0
+  
+  if r <= rc 
+    Δθ          = DF(1//2) 
+  end
+  #Perturbed state:
+  θ            = θ_ref + Δθ # potential temperature
+  π_exner      = DF(1) - grav / (c_p * θ) * x3 # exner pressure
+  ρ            = p0 / (R_gas * θ) * (π_exner)^ (c_v / R_gas) # density
+  P            = p0 * (R_gas * (ρ * θ) / p0) ^(c_p/c_v) # pressure (absolute)
+  T            = P / (ρ * R_gas) # temperature
+  ρu           = SVector(DF(0),DF(0),DF(0))
+  # energy definitions
+  e_kin        = DF(0)
+  e_pot        = grav * x3
+  ρe_tot       = ρ * total_energy(e_kin, e_pot, T)
+  state.ρ      = ρ
+  state.ρu     = ρu
+  state.ρe     = ρe_tot
+  state.moisture.ρq_tot = DF(0)
+end
+# --------------- Driver definition ------------------ # 
+function run(mpicomm, ArrayType, LinearType,
+             topl, dim, Ne, polynomialorder, 
+             timeend, DF, dt)
+  # -------------- Define grid ----------------------------------- # 
+  grid = DiscontinuousSpectralElementGrid(topl,
+                                          FloatType = DF,
+                                          DeviceArray = ArrayType,
+                                          polynomialorder = polynomialorder
+                                           )
+  # -------------- Define model ---------------------------------- # 
+  model = AtmosModel(FlatOrientation(),
+                     HydrostaticState(IsothermalProfile(DF(T_0)),DF(0)),
+                     Vreman{DF}(C_smag),
+                     EquilMoist(), 
+                     NoRadiation(),
+                     Gravity(),
+                     NoFluxBC(),
+                     Initialise_Rising_Bubble!)
+  # -------------- Define dgbalancelaw --------------------------- # 
+  dg = DGModel(model,
+               grid,
+               Rusanov(),
+               CentralNumericalFluxDiffusive(),
+               CentralGradPenalty())
+
+  linmodel = LinearType(model)
+  lindg = DGModel(linmodel,
+               grid,
+               Rusanov(),
+               CentralNumericalFluxDiffusive(),
+               CentralGradPenalty(); auxstate=dg.auxstate)
+
+  Q = init_ode_state(dg, DF(0))
+
+  linearsolver = GeneralizedMinimalResidual(10, Q, 1e-10)
```

Comment:
`1e-10` is too high tolerance for `Float32` . Something like `1e-5` may work in that case.

Reply:
How about?
```suggestion
  linearsolver = GeneralizedMinimalResidual(10, Q, sqrt(eps(DF)))
```

Reply:
Seems fine. The tolerance in the old IMEX examples was set very high to get reproducible test results. In normal runs it will be much smaller. Whatever works to have a test that will not break on different machines is fine here.